### PR TITLE
Issue #515: Take 2, reinstate awaitility

### PR DESCRIPTION
**Pull request by skarsaune**
# Changes

- :bug: Fix bug
Rectify too eager deletion in #515

### CODE REVIEW

**Comment by skarsaune** [comment]
@rhuss  My bad. Deleted one dependency too much now master is broken after #516.

File: client/jmx-adapter/src/test/java/org/jolokia/client/jmxadapter/JmxBridgeTest.java [hunk 1]
```diff
@@ -329,13 +329,14 @@ public void startAgent()
...
-                  }
-                }));
+        		new Callable<Boolean>() {
+
+					@Override
+					public Boolean call() throws Exception {
+						//will throw exception if connection is not ready
+						connector.execute(new J4pVersionRequest());
+						return true;
+					}});
```
Comment (apupier) [inline client/jmx-adapter/src/test/java/org/jolokia/client/jmxadapter/JmxBridgeTest.java:339]: for less change, it is also possible to use:

```
    await()
        .untilAsserted(
                new ThrowingRunnable() {
                  @Override
                  public void run() throws J4pException {
                    connector.execute(new J4pVersionRequest());
                  }
                });
```

but well, it is one line less of code so both solutions seems fine

File: client/jmx-adapter/pom.xml [hunk 2]
```diff
@@ -37,7 +37,12 @@
             <artifactId>testng</artifactId>
             <scope>test</scope>
         </dependency>
-
+        <dependency>
+            <groupId>org.awaitility</groupId>
+            <artifactId>awaitility</artifactId>
+            <version>4.0.0</version>
```
Comment (apupier) [inline client/jmx-adapter/pom.xml:43]: there is a 4.2.0 https://github.com/awaitility/awaitility/releases/tag/awaitility-4.2.0
```suggestion
            <version>4.2.0</version>
```

**Review by rhuss — COMMENTED** [review]
Looks good

**Pull request merged**